### PR TITLE
Support napi inside of bun:ffi

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

This makes it easier to return strings from bun:ffi to JavaScript, by allowing you to go through the napi code path to do so.

Example C:
```c
#include <node/node_api.h>

napi_value napi(napi_env env) {
  napi_value result;
  napi_create_string_utf8(env, "Hello, Napi!", NAPI_AUTO_LENGTH, &result);
  return result;
}
```

JS usage:
```ts
import { cc } from "bun:ffi";
import source from "./hello-napi.c" with { type: "file" };
const {
  symbols: { napi },
} = cc({
  source,
  symbols: {
    "napi": {
      args: ["napi_env"],
      returns: "napi_value",
    },
  },
});

if (napi() !== "Hello, Napi!") {
  throw new Error("napi() !== Hello, Napi!");
}

```

### How did you verify your code works?

Updated a test

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit 820868291037a3facc7c7f12667d244f4bd9a3f1 is building: [`#3597`](https://buildkite.com/bun/bun/builds/3597)

File: test/js/bun/ffi/cc-fixture.js [hunk 1]
```diff
@@ -19,3 +23,7 @@ const {
 if (main() !== 42) {
   throw new Error("main() !== 42");
 }
+
+if (napi_main(null) !== "Hello, Napi!") {
```
Comment (dylan-conway) [inline test/js/bun/ffi/cc-fixture.js:27]: is this going to segfault in the handle scope code?

Comment (Jarred-Sumner) [reply test/js/bun/ffi/cc-fixture.js:27]: discussed IRL

**Comment by codehz** [comment]
is it possible to receive argument via napi?
I mean the "napi_callback_info"

https://www.npmjs.com/package/bigint-buffer <- a 281562 weekly download library is just used for convert between bigint and buffer

**Pull request merged**